datastore/query.go

```diff
@@ -627,6 +629,107 @@ func (c *Client) Count(ctx context.Context, q *Query) (n int, err error) {
 	}
 }
 
+// RunOption lets the user provide options while running a query
+type RunOption interface {
+	apply(*runQuerySettings) error
+}
+
+// ExplainOptions is explain options for the query.
+//
+// Query Explain feature is still in preview and not yet publicly available.
+// Pre-GA features might have limited support and can change at any time.
+type ExplainOptions struct {
+	// When false (the default), the query will be planned, returning only
+	// metrics from the planning stages.
+	// When true, the query will be planned and executed, returning the full
+	// query results along with both planning and execution stage metrics.
+	Analyze bool
+}
```

Comment:
Why is this not structured as a private struct `explainOptions` with a function `ExplainOptions` that returns an instance of `explainOptions`? With this structure the user would have to pass in stuff like `ExplainOptions{Analyze: true}` instead of `ExplainOptions(true)`, right?

Reply:
Right now, ExplainOptions contains only Analyze field. In the future, if more fields are added to ExplainOptions struct, user will no longer be able to use ExplainOptions(true).

Reply:
Ahh I see, I was confused because this is a proto field, not an option in the Golang sense of the word.